src/client/ContentManagement.js

```diff
@@ -621,55 +619,59 @@ exports.CreateContentObject = async function({libraryId, objectId, options={}})
  *
  * @returns {Promise<Object>} - Response containing the object ID and write token of the draft
  */
-exports.CopyContentObject = async function({libraryId, originalVersionHash, options={}, originalObject=null}) {
+exports.CopyContentObject = async function({libraryId, originalVersionHash, options={}}) {
   ValidateLibrary(libraryId);
   ValidateVersion(originalVersionHash);
 
   options.copy_from = originalVersionHash;
 
-  if(!originalObject) {
-    return await this.CreateAndFinalizeContentObject({libraryId, options});
-  }
-
-  ValidateObject(originalObject.id);
+  const {objectId, writeToken} = await this.CreateContentObject({libraryId, options});
+  const originalObjectId = this.utils.DecodeVersionHash(originalVersionHash).objectId;
+  const metadata = await this.ContentObjectMetadata({versionHash: originalVersionHash});
+  const permission = await this.Permission({objectId: originalObjectId});
 
-  const callback = async ({objectId, writeToken}) => {
-    const kmsCapKey = Object.keys(originalObject).find(key => key.startsWith("eluv.caps.ikms"));
-    const userCapKey = Object.keys(originalObject).find(key => key.startsWith("eluv.caps.iusr"));
+  // User CAP
```

Comment:
this can be simplified by just checking the metadata directly, no reason to check if there's any user cap if you already know what you're looking for

```
if(!metadata[userCapKey]) {
  throw Error("...");
}
```